addon/components/uk-modal.js

```diff
@@ -1,58 +1,74 @@
 import { getOwner } from "@ember/application";
 import Component from "@ember/component";
-import { computed } from "@ember/object";
+import { assert } from "@ember/debug";
+import { guidFor } from "@ember/object/internals";
 import { scheduleOnce } from "@ember/runloop";
+import { tracked } from "@glimmer/tracking";
 import UIkit from "uikit";
 
 import layout from "../templates/components/uk-modal";
 
 const noop = () => {};
 
-export default Component.extend({
-  layout,
-
-  modalClass: "",
-  dialogClass: "",
-  btnClose: true,
-  escClose: true,
-  bgClose: true,
-  stack: false,
-  container: true,
-  clsPage: "uk-modal-page",
-  clsPanel: "uk-modal-dialog",
-  selClose: [
+export default class UkModal extends Component {
+  layout = layout;
+
+  modalId = `modal-${guidFor(this)}`;
+
+  @tracked modal;
+  @tracked container = true;
+  @tracked modalClass = "";
+  @tracked dialogClass = "";
+  @tracked btnClose = true;
+  @tracked escClose = true;
+  @tracked bgClose = true;
+  @tracked stack = false;
+  @tracked clsPage = "uk-modal-page";
+  @tracked clsPanel = "uk-modal-dialog";
+  @tracked selClose = [
     ".uk-modal-close",
     ".uk-modal-close-default",
     ".uk-modal-close-outside",
     ".uk-modal-close-full",
-  ].join(", "),
+  ].join(", ");
 
-  isAnimating: false,
+  @tracked isAnimating = false;
+  @tracked focusTrapActive = false;
 
-  modalId: computed("elementId", function () {
-    return `modal-${this.elementId}`;
-  }),
+  modalObserver = null;
 
-  modalHeaderId: computed("modalId", function () {
+  get modalHeaderId() {
     return `${this.modalId}-header`;
-  }),
+  }
 
-  modalSelector: computed("modalId", function () {
+  get modalSelector() {
     return `#${this.modalId}`;
-  }),
+  }
 
-  containerElement: computed("container", function () {
-    return getOwner(this)
-      .lookup("service:-document")
-      .querySelector(this.container);
-  }),
+  get containerSelector() {
+    // Only set the container to the default if no query string was passed as argument.
+    if (typeof this.container !== "string") {
+      return getOwner(this).rootElement || "body";
+    }
+
+    return this.container;
```

Comment:
```suggestion
    return this.args.container;
```
Right?

Reply:
In `@ember/component` the `args` is undefined.

Reply:
Right, in that case I'll resolve those..